src/core/include/openvino/op/constant.hpp

```diff
@@ -843,18 +837,20 @@ class OPENVINO_API Constant : public Op {
     }
 
     size_t mem_size() const {
-        const bool bitwidth_less_than_byte = m_element_type.bitwidth() < 8;
-        if (bitwidth_less_than_byte) {
-            const auto size = shape_size(m_shape);
-            const auto bitwidth = size * m_element_type.bitwidth();
-            // for rounding by `(bitwidth + 7) / 8` will work for
-            // `bitwidth < numeric_limits<size_t>::max() - 7`
-            return bitwidth / 8 + (bitwidth % 8 ? 1 : 0);
+        constexpr size_t bits_in_byte = 8;
+        const auto bit_width = m_element_type.bitwidth();
+        const auto bit_size = shape_size(m_shape) * bit_width;
+        if ((bit_width < bits_in_byte) && (bit_size % bits_in_byte)) {
+            return (bit_size / bits_in_byte) + 1;
+        } else {
+            return bit_size / bits_in_byte;
         }
-        return shape_size(m_shape) * m_element_type.size();
```

Comment:
Looks like `m_element_type.size()` not equal to `m_element_type.bitwidth()`, so this port isn't equal to previous

Example:
```
m_element_type.bitwidth() = 9
shape_size(m_shape) = 8
bit_size / bits_in_byte == 8 * 9 / 8 == 9
```
but
`shape_size(m_shape) * m_element_type.size() == 8 * ((9 + 7) >> 3) == 8 * 2 == 16`

Am I correct?

Reply:
Right now `bitwidths` defined for all precision are power of 2 values. So new version will give same result as old one.

For bit-width `9` there will be difference between old and new version. 
But make calculation in old way there is assumption for 9-bits that values will stored on two bytes with gaps, but maybe it should stored without gaps.

I can restore old version for bit-widths greater/equal 8 but not sure if this calculation is correct.

for new version the condition `if ((bit_width < bits_in_byte) && (bit_size % bits_in_byte))` I thinks should be  replaced by `if (bit_size % bits_in_byte)` then additional byte will be added into count if required.

If you know which approach is correct (store 9-bit with gaps or without) then will apply the change or restore the previous one.

Reply:
`mem_size` is used in Constant `fill_data` to fill `mem_size()` elements starting from some pointer (for low precision types it will be char* or int8_t*). So, `mem_size()` should return value aligned by 8 (byte size)

Reply:
I've restored previous version for `mem_size` calculation for `bit-witdth >= 8`.